tournament/src/api/views.py

```diff
@@ -0,0 +1,128 @@
+from django.http import JsonResponse
+from django.utils.decorators import method_decorator
+from django.views import View
+from django.views.decorators.csrf import csrf_exempt
+from django.forms.models import model_to_dict
+
+from dateutil import parser, tz
+from datetime import datetime, timezone
+import json
+
+from api.models import Tournament
+from tournament.authenticate_request import authenticate_request
+from tournament import settings
+
+
+@method_decorator(csrf_exempt, name='dispatch')
+class TournamentView(View):
+    @staticmethod
+    def post(request):
+        try:
+            user, authenticate_errors = authenticate_request(request)
+            if user is None:
+                return JsonResponse(data={'errors': authenticate_errors}, status=401)
+
+            json_request = json.loads(request.body.decode('utf8'))
+            valid_tournament, errors = TournamentView.is_valid_tournament(json_request)
+            if not valid_tournament:
+                return JsonResponse(data={'errors': errors}, status=400)
+
+            tournament = Tournament(
+                name=json_request['name'],
+                is_private=json_request['is-private'],
+                admin_id=user['id']
+            )
+            max_players = json_request.get('max-players')
+            if max_players is not None:
+                tournament.max_players = max_players
+            registration_deadline = json_request.get('registration-deadline')
+            if json_request.get('registration-deadline') is not None:
+                registration_deadline = parser.isoparse(registration_deadline)
+                tournament.registration_deadline = TournamentView.convert_to_utc_datetime(registration_deadline)
+            tournament.save()
+
+            return JsonResponse(model_to_dict(tournament), status=201)
+        except json.JSONDecodeError:
+            return JsonResponse(data={'errors': ['Invalid JSON format in request body']}, status=400)
```

Comment:
Why do everything in the try-except block? Personally, that's something I dislike about try-except; it feels like we end up doing everything inside it. I know it's debatable, but what do you think about this alternative?


```
def post(request):
    user, authenticate_errors = authenticate_request(request)
    if user is None:
        return JsonResponse(data={'errors': authenticate_errors},
                            status=401)
    try:
        json_request = json.loads(request.body.decode('utf8'))
    except json.JSONDecodeError:
        return JsonResponse(
            data={'errors': ['Invalid JSON format in request body']},
            status=400)
    valid_tournament, errors = TournamentView.is_valid_tournament(
        json_request)
    if not valid_tournament:
        return JsonResponse(data={'errors': errors}, status=400)

    tournament = Tournament(
        name=json_request['name'],
        is_private=json_request['is-private'],
        admin_id=user['id']
    )
    max_players = json_request.get('max-players')
    if max_players is not None:
        tournament.max_players = max_players
    registration_deadline = json_request.get('registration-deadline')
    if json_request.get('registration-deadline') is not None:
        registration_deadline = parser.isoparse(registration_deadline)
        tournament.registration_deadline = TournamentView.convert_to_utc_datetime(
            registration_deadline)
    tournament.save()
    return JsonResponse(model_to_dict(tournament), status=201)

```

Reply:
that's fine for me. I didn't know that json_request remained available outside the scope of the try block.

Reply:
87c7dc3c4536ca5c9b84a5a8aeb10a5f2c05b071